### PR TITLE
Do not skip on FuncCall

### LINKED ISSUE #21: Class used in the same namespace but marked as unused
Hi, @TomasVotruba thanks for creating this tool. We started using it in our company, and it found many classes that can be removed! We love it!

To the point, we found this tool has problems when a class is used in the same namespace. It marks it as unused, but in fact, a class is used.

Example: 
```php
<?php

namespace App;

class A {
    public function main()
    {
        return B::someMethod();
    }
}
```

```php
<?php

namespace App;

class B {
    public static function someMethod()
    {
        return 1;
    }
}
```

In this example, class B is marked as unused. To fix it, I have to add `use`, but in this case it is redundant. Rule **no_unused_imports** in phpcs remove it.

```diff
<?php

namespace App;

+use App/B;

class A {
    public function main()
    {
        return B::someMethod();
    }
}
```

and then class-leak tool doesn't mark class `B` as unused. 

I haven't taken a look in codebase deeply, but I would love to introduce a fix for that, but I want to hear that it is possible/doable without refactoring a whole tool :D

### PR DESCRIPTION
**Pull request by ruudk**
The call could be a static call, in that case, you do want to traverse.

Fixes #21

### CODE REVIEW

File: src/NodeVisitor/UsedClassNodeVisitor.php [hunk 1]
```diff
@@ -33,10 +33,6 @@ public function beforeTraverse(array $nodes): array
 
     public function enterNode(Node $node): Node|null|int
     {
-        if ($node instanceof FuncCall) {
```
Comment (samsonasik) [inline src/NodeVisitor/UsedClassNodeVisitor.php:36]: Should be changed to $node instanceof Class_ || $node instanceof Function_ || $node instanceof Closure

Comment (samsonasik) [reply src/NodeVisitor/UsedClassNodeVisitor.php:36]: hmm, it seems scan all stmts, It seems need test to cover it tho

**Comment by samsonasik** [comment]
@ruudk could you add failing test case?

**Comment by ruudk** [comment]
@samsonasik Thanks, I wrote a test and then came up with a different solution that's better.

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @ruudk , I don't have access to this repo so I will let @TomasVotruba decide ;)

File: tests/UseImportsResolver/UseImportsResolverTest.php [hunk 2]
```diff
@@ -34,6 +36,6 @@ public function test(string $filePath, array $expectedClassUsages): void
 
     public static function provideData(): Iterator
     {
-        yield [__DIR__ . '/Fixture/FileUsingOtherClasses.php', [FirstUsedClass::class, SecondUsedClass::class]];
+        yield [__DIR__ . '/Fixture/FileUsingOtherClasses.php', [SomeFactory::class, FirstUsedClass::class, FourthUsedClass::class, SecondUsedClass::class]];
```
Comment (TomasVotruba) [inline tests/UseImportsResolver/UseImportsResolverTest.php:39]: Could you create a separate fixture file? Just so we have this clearly separated.

Comment (ruudk) [reply tests/UseImportsResolver/UseImportsResolverTest.php]: Good idea, done!

**Comment by TomasVotruba** [comment]
Looks good :+1:  Thank you

**Pull request merged**